<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [1]:
# Membuat dataframes dari dictionary
import pandas as pd

def data(N):
    nama = []
    umur = []
    for i in range(N):
        nama.append(input('Masukan nama: '))
        umur.append(int(input('Masukan umur: ')))
    dict = {
        'Name': nama, 
        'Age': umur
    }
    return pd.DataFrame(dict)

print(data(int(input('Banyak data: '))))

Banyak data: 1
Masukan nama: Steven
Masukan umur: 18
     Name  Age
0  Steven   18


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [2]:
# Membaca data dan memasukannya kedalam suatu variabel
import pandas as pd
df = pd.read_csv('diamonds.csv')
print(df)

       Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]


Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab:
-Perbedaan series dengan list adalah jika diprint series akan dicetak bersamaan dengan indexnya sedangkan list tidak. Kedua, index series tidak harus berupa integer, sedangkan index dari list adalah integer.

Perbedaan series dengan dictionary adalah series bisa mempunyai key/index yang duplikat dan series juga memiliki kemampuan untuk slicing tidak seperti dictionary

-Perbedaan data frame dan series adalah dimana data frame memiliki 2 dimensi atau seperti tabel sedangkan series hanya 1 dimensi

In [3]:
# code here
pd.Series([1,3,4,2])

0    1
1    3
2    4
3    2
dtype: int64

<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [4]:
# code here
print(df.price[0])
print(df.color[len(df)-1])

326
D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: Deskriptif statistik adalah metode yang berkaitan dengan menjelaskan dan merumuskan suatu data

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [5]:
# code here
df.describe()

Unnamed: 0         carat  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean   26970.500000      0.797940  ...      5.734526      3.538734
std    15571.281097      0.474011  ...      1.142135      0.705699
min        1.000000      0.200000  ...      0.000000      0.000000
25%    13485.750000      0.400000  ...      4.720000      2.910000
50%    26970.500000      0.700000  ...      5.710000      3.530000
75%    40455.250000      1.040000  ...      6.540000      4.040000
max    53940.000000      5.010000  ...     58.900000     31.800000

[8 rows x 8 columns]

Jawab:
nilai rata-rata dari column price adalah 3932.799722
nilai standar deviasi dari column depth adalah 1.432621
nilai maximum dari column carat adalah 5.010000